json-logic/schema.ts

```diff
@@ -0,0 +1,146 @@
+// LogicExpression refers to the JSONLogic standard.
+import { LogicExpression } from "./logic";
+
+/*
+* This module defines a schema for questionaire and result calulation logic.
+*/
+
+/**
+ * Type of a question, essentially defines the result type.
+ */
+export enum QuestionType {
+	Select = "Select",
+	Multiselect = "Multiselect",
+	Number = "Number",
+	Boolean = "Boolean",
+	Date = "Date",
+	Text = "Text"
+}
+
+/**
+ * Option for multi-select questions.
+ */
+export interface IOption {
+	/**
+	 * Human-Readable answer, can be localized.
+	 */
+	text: string,
+	/**
+	 * Value used for evaluating logic expressions.
+	 */
+	value: string;
+}
+
+/**
+ * Represents a question.
+ */
+export interface IQuestion {
+	/**
+	 * Unique id for referring this question in logic expressions.
+	 */
+	id: string
+	/**
+	 * Type of the question.
+	 */
+	type: QuestionType,
+	/**
+	 * Human-readable question text, can be localized.
+	 */
+	text: string,
+	/**
+	 * Answer options for Select/Multiselect questions.
+	 */
+	options?: IOption[],
+	/**
+	 * Boolean indicating whether the question is optional or not.
+	 */
+	optional?: boolean
+	/**
+	 * Logic expression to decide whether the question should be displayed or not.
+	 */
+	skipIf?: LogicExpression
+}
+
+/**
+ * Represents a variable which is computed from the given answers or other variables.
+ */
+export interface IVariable {
+	/**
+	 * Unique id for referring this variable in logic expressions.
+	 */
+	id: string
+	/**
+	 * Logic expression used to compute this variable.
+	 */
+	value: LogicExpression;
+}
+
+/**
+ * Represents a result category. A category might yield exactly one or zero results at the end of the questionaire.
+ */
+export interface IResultCategory {
```

Comment:
:+1:  for this idea!

Reply:
Could you try to formulate it for the current example?

I'm afraid it's harder to explain though, but I also agree the current solution is not 100% optinal.